### PR TITLE
Add foreign key to ownership model

### PR DESCRIPTION
**Pull request by kevinlinxc**
Follow-up to https://github.com/rubygems/rubygems.org/pull/3091, addressing https://app.honeybadger.io/projects/40972/faults/68202479

Adds foreign key to ownership db object, so that when a user is deleted, all their related ownerships get deleted too

### CODE REVIEW

File: db/migrate/20220614221414_ownership_foreign_key.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class OwnershipForeignKey < ActiveRecord::Migration[7.0]
+  def up
+    add_foreign_key :ownerships, :users, on_delete: :cascade
+  end
+
+  def gem_downloads
```
Comment (deivid-rodriguez) [inline db/migrate/20220614221414_ownership_foreign_key.rb:6]:
```suggestion
  def down
```

🤔

Comment (kevinlinxc) [reply db/migrate/20220614221414_ownership_foreign_key.rb:6]: autocomplete got me good

**Comment by kevinlinxc** [comment]
My testing
1. Run elasticsearch, memcached, postgresql
2. Seed database with gems and owners https://gist.github.com/kevinlinxc/82044f2674fbc4faf2c517e825b39818
3. Find an owner, and their ownerships:
image
4. Make sure the migration is run

5. Destroy the owner:
`User.where(id:7).destroy_all`

(Took >30 minutes)
6. Check that their ownerships are deleted too:
image

**Review by simi — APPROVED** [review]

**Comment by simi** [comment]
@sonalkr132 safe to deploy from your side? I can also deploy change in DB manually using `NOT VALID` and `VALIDATE` trick to deploy with no table blocking.

**Comment by sonalkr132** [comment]
We need a way to dry run this first. Or please verify manually what excatly is this going to delete.

**Review by sonalkr132 — APPROVED** [review]

**Pull request closed without merging**